File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findDuplicates(int* nums, int numsSize, int* returnSize) {
    
    int *resultset=(int*)malloc(numsSize*sizeof(int));
    int size=0;
    
    for(int i=0;i<numsSize;i++){
        int index=abs(nums[i])-1;
        if(nums[index]<0){
            resultset[size++]=abs(nums[i]);
        }
        nums[index]=-1*nums[index];
    }
    
    *returnSize=size;
    return resultset;
    
}

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        ArrayList<Integer> al=new ArrayList<>();
        for(int i=0;i<nums.length;i++){
            int index=Math.abs(nums[i])-1;
            if(nums[index]<0){
                al.add(Math.abs(nums[i]));
            }
            else{
                nums[index]=-1*nums[index];
            }
        }
        return al;
    }
}